lib/utils.py

```diff
@@ -34,19 +34,50 @@ def build_architecture(arch, cross, num_cpus, target):
 def build_check(target: Path) -> bool:
     '''Checks if the linux kernel builds properly'''
 
+    specific_arch = None
+    path_parts = target.parts
+    if 'arch' in path_parts:
+        arch_index = path_parts.index('arch')
+        specific_arch = path_parts[arch_index + 1]
+
     num_cpus = len(os.sched_getaffinity(0))
     shutil.copy('.config', '.tmp.config')
     try:
         # TODO: S390 has Clang support but for Clang-15, Debian currently supports Clang-14
         clang_archs = ["arm", "arm64", "mips", "riscv", "powerpc", "x86", "um"]
         for arch in clang_archs:
+            if specific_arch and specific_arch != arch:
+                continue
             build_architecture(arch, 'LLVM=1', num_cpus, target)
 
         # Architecture crosstools found here:  https://mirrors.edge.kernel.org/pub/tools/crosstool
         architectures = ['alpha', 'arc', 'csky', 'hppa', 'hppa64', 'i386', 'loongarch64', 'm68k', 'microblaze',
                          'mips64', 'nios2', 'or1k', 's390', 'sh2', 'sh4', 'sparc', 'sparc64', 'xtensa']
 
+        name_mappings = {
+            'alpha': 'alpha',       # DEC Alpha
+            'arc': 'arc',           # Argonaut RISC Core
+            'csky': 'csky',         # C-SKY
+            'hppa': 'parisc',       # HP Precision Architecture
+            'hppa64': 'parisc',   # HP Precision Architecture (64-bit)
+            'i386': 'x86',          # Intel 386 (often grouped under 'x86')
+            'loongarch64': 'loongarch', # LoongArch 64-bit
+            'm68k': 'm68k',         # Motorola 68000 series
+            'microblaze': 'microblaze', # Xilinx MicroBlaze
```

Comment:
If we're going to align most of the comments, then let's align 100% of them (parisc, loongarch, microblaze).